src/app/_helpers/string-helpers.ts

```diff
@@ -68,35 +69,82 @@ export function rightsUrlMatch(url: string): string | null {
   return null;
 }
 
+/** sanitiseSearchTerm
+ * incorporates diaritics, escapes regex operators / qualifiers and
+ * (conditionally) applies start/end modifiers
+ *
+ * @param {string} filterString - filter information
+ * @returns the string converted to a regex-safe string
+ **/
+export function sanitiseSearchTerm(filterString: string): string {
+  if (
+    filterString.length === 0 ||
+    filterString.replace(sanitationRegex, '').length === 0
+  ) {
+    return '';
+  }
+
+  const modifierStart = filterString.indexOf('^') === 0;
+  const modifierEnd = filterString.indexOf('$') === filterString.length - 1;
+
+  if (modifierStart) {
+    filterString = filterString.slice(1);
+  }
+  if (modifierEnd) {
+    filterString = filterString.slice(0, -1);
+  }
+
+  let filter = appendDiacriticEquivalents(
+    replaceDiacritics(filterString.replace(sanitationRegex, sanitationReplace))
+  );
+
+  if (modifierStart && modifierEnd) {
+    filter = `^${filter}$|\\^${filter}$|^${filter}\\$|\\^${filter}\\$`;
+  } else if (modifierStart) {
+    filter = `^${filter}|\\^${filter}`;
+  } else if (modifierEnd) {
+    filter = `${filter}$|${filter}\\$`;
+  }
+
+  return filter;
+}
+
 /** filterList
-/* @returns filterables filtered by filterString
-**/
+ *
+ * @param {string} filterString - filter information
+ * @param {Array<T>} filterables - the array to filter
+ * @param {string?} filterProp - the object property to filter on
+ *
+ * @returns filterables filtered by filterString
+ **/
 export function filterList<T>(
   filterString: string,
   filterables: Array<T>,
   filterProp?: string
 ): Array<T> {
-  if (invalidRegexes.includes(filterString)) {
-    return [];
+  if (filterString.length === 0) {
+    return filterables;
   }
 
-  const filter = replaceDiacritics(filterString);
-  const reg = new RegExp(appendDiacriticEquivalents(filter), 'gi');
-  return filterables.filter((filterable: unknown) => {
+  let filter = sanitiseSearchTerm(filterString);
+  console.log(filter + ' ' + filter.length + ' ' + filterString);
```

Comment:
It seems this was used for debugging. Is it still required ? Could it be commented, for example?

Reply:
Deleted!